client/src/components/Sidebar/index.tsx

```diff
@@ -20,6 +21,12 @@ export function initSidebar() {
   if (element instanceof HTMLElement && element.dataset.props) {
     const props = window.telepath.unpack(JSON.parse(element.dataset.props));
 
+    const preferences: SidebarPreferencesDefinition = props.preferences;
+
+    if (preferences.collapsed) {
+      document.body.classList.add('sidebar-collapsed');
```

Comment:
There is a slight issue with this code: it can sometimes run a bit late. This means that when the sidebar is collapsed and the page is loaded, a transition is visible where the main content expands to fill the whitespace left by the collapsed sidebar. This looks a bit odd and can be fixed by doing this check inside the Django template, not here in the frontend.